httphandler/response_merger.go

```diff
@@ -36,6 +38,22 @@ func extractDestinationHostName(r transport.ResErrTuple) string {
 	return ""
 }
 
+func extractAccessKey(req *http.Request) string {
```

Comment:
Add a comment here. I do understand what is happening but I do not understand why